<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Data preparation examples &mdash; Omics Playground  documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
    <link rel="shortcut icon" href="../_static/bigomics-logo-white-48px.png"/>
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Manual to prepare the samples.csv file" href="samples.html" />
    <link rel="prev" title="Data import and precomputation" href="dataprep.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            Omics Playground
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../overview.html">Overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="../install.html">Installation</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Data Preparation</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="dataprep.html">Data import and precomputation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Data preparation examples</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#uploading-data-through-the-interface">Uploading data through the interface</a></li>
<li class="toctree-l2"><a class="reference internal" href="#preparing-an-input-object-using-scripts">Preparing an input object using scripts</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#from-fastq-files">From FASTQ files</a></li>
<li class="toctree-l3"><a class="reference internal" href="#from-gene-counts-table-or-geo-repository">From gene counts table or GEO repository</a></li>
<li class="toctree-l3"><a class="reference internal" href="#from-single-cell-data">From single-cell data</a></li>
<li class="toctree-l3"><a class="reference internal" href="#from-lc-ms-ms-proteomics-data">From LC-MS/MS proteomics data</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="samples.html">Manual to prepare the <cite>samples.csv</cite> file</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Analysis Modules</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../modules/outline.html">Outline</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/mod0_home.html">Home Panel</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/mod1_dataview.html">Data View</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/mod2_clustering.html">Clustering</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/mod3_DEGs.html">Expression</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/mod4_DEGSs.html">Enrichment</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/mod7_Signature.html">Signature</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/mod9_CellProfiling.html">Cell Profiling</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Examples</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../examples/examples.html">Reanalyzing Public Datasets</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Methods</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../methods.html">Methods</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">FAQ</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../faq.html">Frequently Asked Questions (FAQ)</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">About</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../info.html">Contact</a></li>
<li class="toctree-l1"><a class="reference internal" href="../info.html#citation">Citation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../info.html#license">License</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Omics Playground</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Data preparation examples</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/dataprep/dataprep_example.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="data-preparation-examples">
<span id="dataprep-example"></span><h1>Data preparation examples<a class="headerlink" href="#data-preparation-examples" title="Permalink to this headline">¶</a></h1>
<p>The platform requires the transcriptomics and proteomics data to be in a
structured format as an input. Users can inject their data to the platform by
either <a class="reference internal" href="#uploading-through-the-interface">uploading through the interface</a> or
<a class="reference internal" href="#id1">preparing an input object using scripts</a>.</p>
<section id="uploading-data-through-the-interface">
<h2>Uploading data through the interface<a class="headerlink" href="#uploading-data-through-the-interface" title="Permalink to this headline">¶</a></h2>
<p id="uploading-through-the-interface">Users can import their data from the <strong>Upload data</strong> panel located under the
<a class="reference internal" href="../modules/mod0_home.html#home"><span class="std std-ref">Home Panel</span></a> module. The platform requires the list of files as shown in the
figure below.
The format of files must be comma-separated-values (CSV) text.
It is important to name the files exactly as shown.
Be sure the dimensions, rownames and column names match for all files.
It is required to provide a unique name and brief description for the data set
while uploading.</p>
<dl class="field-list simple">
<dt class="field-odd"><strong>counts.csv</strong></dt>
<dd class="field-odd"><p>Count/expression file with gene on rows, samples as columns.</p>
</dd>
<dt class="field-even"><strong>samples.csv</strong></dt>
<dd class="field-even"><p>Samples file with samples on rows, phenotypes as columns.</p>
</dd>
<dt class="field-odd"><strong>contrasts.csv</strong></dt>
<dd class="field-odd"><p>Contrast file with samples on rows, conditions as columns.</p>
</dd>
</dl>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc1.3.png"><img alt="../_images/psc1.3.png" src="../_images/psc1.3.png" style="width: 100%;" /></a>
</figure>
</section>
<section id="preparing-an-input-object-using-scripts">
<h2>Preparing an input object using scripts<a class="headerlink" href="#preparing-an-input-object-using-scripts" title="Permalink to this headline">¶</a></h2>
<p id="id1">Alternatively, an input object can be prepared by using scripts with more
detailed data cleaning, filtering, normalisation and preprocessing.
Users can prepare an input data from
their own FASTQ files, gene counts tables, or from a dataset of interest stored
in public repositories such as <a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/">GEO</a>.
Similarly, they can also prepare an input from single-cell or LC-MS/MS proteomics
datasets.
We provide four types of example cases to guide users for preparing their input
objects and injecting it into the platform. Basically, the example cases
illustrate how to prepare an input data:</p>
<ol class="arabic simple">
<li><p><a class="reference internal" href="#id2">from FASTQ files</a>,</p></li>
<li><p><a class="reference internal" href="#from-gene-counts-table-or-from-the-geo-repository">from gene counts table or from the GEO repository</a>,</p></li>
<li><p><a class="reference internal" href="#id3">from single-cell data</a>,</p></li>
<li><p><a class="reference internal" href="#id4">from LC-MS/MS proteomics data</a>.</p></li>
</ol>
<p>All the necessary scripts for data cleaning and preprocessing examples can be
found under the <code class="docutils literal notranslate"><span class="pre">scripts/</span></code> folder.</p>
<section id="from-fastq-files">
<h3>From FASTQ files<a class="headerlink" href="#from-fastq-files" title="Permalink to this headline">¶</a></h3>
<p id="id2">Starting from FASTQ files, we recommend using the <a class="reference external" href="https://www.rdocumentation.org/packages/GREP2/versions/1.0.2">GREP2</a>
package to obtain gene counts through quality control, trimming,
quantification of gene abundance, and so on.  Afterwards, the user can
refer to the examples in the next section for preparing an input data
from the gene counts.</p>
</section>
<section id="from-gene-counts-table-or-geo-repository">
<h3>From gene counts table or GEO repository<a class="headerlink" href="#from-gene-counts-table-or-geo-repository" title="Permalink to this headline">¶</a></h3>
<p id="from-gene-counts-table-or-from-the-geo-repository">Users can prepare an input data from their own gene counts or download a
relevant dataset from repositories such as <a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/">GEO</a>.
Some examples are provided in the following scripts:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://portal.gdc.cancer.gov/projects/TCGA-BRCA">TCGA-BRCA</a>: <code class="docutils literal notranslate"><span class="pre">pgx-tcga-brca.R</span></code></p></li>
<li><p><a class="reference external" href="https://www.cbioportal.org/study/summary?id=prad_tcga">TCGA-PRAD</a>: <code class="docutils literal notranslate"><span class="pre">pgx-tcga-prad.R</span></code></p></li>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE10846">GSE10846</a>: <code class="docutils literal notranslate"><span class="pre">pgx-GSE10846-dlbcl.R</span></code></p></li>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE114716">GSE114716</a>: <code class="docutils literal notranslate"><span class="pre">pgx-GSE114716-ipilimumab.R</span></code></p></li>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE22886">GSE22886</a>: <code class="docutils literal notranslate"><span class="pre">pgx-GSE22886-immune.R</span></code></p></li>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE28492">GSE28492</a>: <code class="docutils literal notranslate"><span class="pre">pgx-GSE28492-roche.R</span></code></p></li>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE32591">GSE32591</a>: <code class="docutils literal notranslate"><span class="pre">pgx-GSE32591-lupusnephritis.R</span></code></p></li>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE53784">GSE53784</a>: <code class="docutils literal notranslate"><span class="pre">pgx-GSE53784-wnvjev.R</span></code></p></li>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE88808">GSE88808</a>: <code class="docutils literal notranslate"><span class="pre">pgx-GSE88808-prostate.R</span></code></p></li>
</ul>
</section>
<section id="from-single-cell-data">
<h3>From single-cell data<a class="headerlink" href="#from-single-cell-data" title="Permalink to this headline">¶</a></h3>
<p id="id3">Single-cell RNA sequencing experiments have been valuable to provide insights into
complex biological systems, reveal complex and rare cell populations, uncover
relationships between genes, and track the trajectories of cell lineages.
Below we provide some data preparation examples from single-cell experiments:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE72056">GSE72056</a>: <code class="docutils literal notranslate"><span class="pre">pgx-GSE72056-scmelanoma.R</span></code></p></li>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE92332">GSE92332</a>: <code class="docutils literal notranslate"><span class="pre">pgx-GSE92332-scintestine.R</span></code></p></li>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE98638">GSE98638</a>: <code class="docutils literal notranslate"><span class="pre">pgx-GSE98638-scliver.R</span></code></p></li>
</ul>
</section>
<section id="from-lc-ms-ms-proteomics-data">
<h3>From LC-MS/MS proteomics data<a class="headerlink" href="#from-lc-ms-ms-proteomics-data" title="Permalink to this headline">¶</a></h3>
<p id="id4">Two examples are provided below for LC-MS/MS proteomics data preprocessing:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/27745970">Geiger et al. 2016</a>: <code class="docutils literal notranslate"><span class="pre">pgx-geiger2016-arginine.R</span></code></p></li>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/28263321">Rieckmann et al. 2017</a>: <code class="docutils literal notranslate"><span class="pre">pgx-rieckmann2017-immprot.R</span></code></p></li>
</ul>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="dataprep.html" class="btn btn-neutral float-left" title="Data import and precomputation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="samples.html" class="btn btn-neutral float-right" title="Manual to prepare the samples.csv file" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright .</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>