<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Loading custom geneset files &mdash; Omics Playground  documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
    <link rel="shortcut icon" href="../_static/bigomics-logo-white-48px.png"/>
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Data processing inside Omics Playground" href="dataprocess.html" />
    <link rel="prev" title="Preparing input files from raw data formats" href="dataprep_example.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            Omics Playground
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../overview.html">Overview</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Basic Data Preparation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="dataprep.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="samples.html">Samples file</a></li>
<li class="toctree-l1"><a class="reference internal" href="counts.html">Counts file</a></li>
<li class="toctree-l1"><a class="reference internal" href="contrasts.html">Contrasts file (optional)</a></li>
<li class="toctree-l1"><a class="reference internal" href="datachecks.html">Checks before uploading</a></li>
<li class="toctree-l1"><a class="reference internal" href="uploadopg.html">Uploading your data in Omics Playground</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Advanced Data Preparation</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="computepgx.html">What if I have many datasets?</a></li>
<li class="toctree-l1"><a class="reference internal" href="dataprep_example.html">Preparing input files from raw data formats</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Loading custom geneset files</a></li>
<li class="toctree-l1"><a class="reference internal" href="dataprocess.html">Data processing inside Omics Playground</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Analysis Modules</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../modules/outline.html">Outline</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/mod0_home.html">Home Panel</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/mod1_dataview.html">Data View</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/mod2_clustering.html">Clustering</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/mod3_DEGs.html">Expression</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/mod4_DEGSs.html">Enrichment</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/mod7_Signature.html">Signature</a></li>
<li class="toctree-l1"><a class="reference internal" href="../modules/mod9_CellProfiling.html">Cell Profiling</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Examples</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../examples/examples.html">Reanalyzing Public Datasets</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Methods</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../methods.html">Methods</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">FAQ</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../faq.html">Frequently Asked Questions (FAQ)</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">About</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../info.html">Contact</a></li>
<li class="toctree-l1"><a class="reference internal" href="../info.html#citation">Citation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../info.html#license">License</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Omics Playground</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Loading custom geneset files</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/dataprep/geneset.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="loading-custom-geneset-files">
<span id="geneset"></span><h1>Loading custom geneset files<a class="headerlink" href="#loading-custom-geneset-files" title="Permalink to this headline">Â¶</a></h1>
<p>GMT (Gene Matrix Transposed) files are a format commonly used
in bioinformatics and gene set enrichment analysis (GSEA).</p>
<p>Gene sets are collections of genes that share certain biological
characteristics, such as being involved in the same biological
pathway or having similar functions. GSEA is a computational
method used to determine whether a particular gene set is
significantly enriched in a given dataset of gene expression.</p>
<p>In a GMT file, each line represents a gene set and consists
of a gene set name, and the list of genes
belonging to that set. The gene names are typically represented
using gene symbols or other identifiers.</p>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>If you want to create your own genesets, you can follow the tutorials in R
<a class="reference external" href="https://rdrr.io/cran/ActivePathways/man/GMT.html">here</a> or
more information about the GMT format <a class="reference external" href="https://www.gsea-msigdb.org/gsea/doc/GSEAUserGuideTEXT.htm#_Gene_Sets_File_GMT">here</a>.</p>
</div>
<p>In Omics playground, GMT files are the backbone of several modules,
where we aim to identify enriched gene sets associated with specific
biological processes, diseases, or experimental conditions.</p>
<p>You can find the geneset availables in omicsplayground with the R package playbase.
First, install playbase with <code class="docutils literal notranslate"><span class="pre">devtools::install_github(&quot;omicsplayground/playdata&quot;)</span></code>
and then run the command <code class="docutils literal notranslate"><span class="pre">playdata::GSET_INFO</span></code>.</p>
<p>If you have your own geneset, or you want to include a GMT file from a database,
you can upload directly in the Upload module under Options.</p>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p>If you are working in R, you can add your gmt file to the <code class="docutils literal notranslate"><span class="pre">playbase::pgx.computePGX</span></code> by providing the <code class="docutils literal notranslate"><span class="pre">custom.geneset</span></code> argument.
custom.geneset argument. That should be a list where the first argument <code class="docutils literal notranslate"><span class="pre">gmt</span></code> is the gmt file (see <code class="docutils literal notranslate"><span class="pre">playbase::EXAMPLE_GMT</span></code>), and the second argument is the name of the gmt file.</p>
<p>Building on what we learned in the section <a class="reference internal" href="computepgx.html#computepgx"><span class="std std-ref">Computing PGX</span></a>
, we can add our own geneset to the analysis with the followign call:</p>
<div class="highlight-R notranslate"><div class="highlight"><pre><span></span><span class="n">filePath</span> <span class="o">&lt;-</span> <span class="nf">getwd</span><span class="p">()</span> <span class="c1"># path to gmt file</span>

<span class="n">custom.geneset</span> <span class="o">&lt;-</span> <span class="nf">list</span><span class="p">()</span>

<span class="n">custom.geneset</span><span class="o">$</span><span class="n">gmt</span> <span class="o">&lt;-</span> <span class="n">playbase</span><span class="o">::</span><span class="nf">read.gmt</span><span class="p">(</span><span class="n">filePath</span><span class="p">)</span>

<span class="c1"># perform some basic checks</span>
<span class="n">gmt.length</span> <span class="o">&lt;-</span> <span class="nf">length</span><span class="p">(</span><span class="n">custom.geneset</span><span class="o">$</span><span class="n">gmt</span><span class="p">)</span>
<span class="n">gmt.is.list</span> <span class="o">&lt;-</span> <span class="nf">is.list</span><span class="p">(</span><span class="n">custom.geneset</span><span class="o">$</span><span class="n">gmt</span><span class="p">)</span>

<span class="c1"># clean genesets</span>

<span class="n">custom.geneset</span><span class="o">$</span><span class="n">gmt</span> <span class="o">&lt;-</span> <span class="nf">list</span><span class="p">(</span><span class="n">CUSTOM</span> <span class="o">=</span> <span class="n">custom.geneset</span><span class="o">$</span><span class="n">gmt</span><span class="p">)</span>

<span class="c1"># convert gmt to OPG standard</span>
<span class="n">custom.geneset</span><span class="o">$</span><span class="n">gmt</span> <span class="o">&lt;-</span> <span class="n">playbase</span><span class="o">::</span><span class="nf">clean_gmt</span><span class="p">(</span><span class="n">custom.geneset</span><span class="o">$</span><span class="n">gmt</span><span class="p">,</span><span class="s">&quot;CUSTOM&quot;</span><span class="p">)</span>

<span class="c1"># compute custom geneset stats</span>
<span class="n">custom.geneset</span><span class="o">$</span><span class="n">gmt</span> <span class="o">&lt;-</span> <span class="n">custom.geneset</span><span class="o">$</span><span class="n">gmt</span><span class="p">[</span><span class="o">!</span><span class="nf">duplicated</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">custom.geneset</span><span class="o">$</span><span class="n">gmt</span><span class="p">))]</span>
<span class="n">custom.geneset</span><span class="o">$</span><span class="n">info</span><span class="o">$</span><span class="n">GSET_SIZE</span> <span class="o">&lt;-</span> <span class="nf">sapply</span><span class="p">(</span><span class="n">custom.geneset</span><span class="o">$</span><span class="n">gmt</span><span class="p">,</span><span class="n">length</span><span class="p">)</span>

<span class="c1"># pass the custom.geneset as argument to pgx.computePGX</span>
<span class="n">pgx</span> <span class="o">&lt;-</span> <span class="n">playbase</span><span class="o">::</span><span class="nf">pgx.computePGX</span><span class="p">(</span>
    <span class="n">pgx</span> <span class="o">=</span> <span class="n">pgx</span><span class="p">,</span>
    <span class="n">custom.geneset</span> <span class="o">=</span> <span class="n">custom.geneset</span>
<span class="p">)</span>
</pre></div>
</div>
<p>Currently, we are supporting only 1 gmt file in the list, but we may implement multiple gmt files in the future.</p>
</div>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="dataprep_example.html" class="btn btn-neutral float-left" title="Preparing input files from raw data formats" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="dataprocess.html" class="btn btn-neutral float-right" title="Data processing inside Omics Playground" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright .</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>