<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Home Panel &mdash; Omics Playground  documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
    <link rel="shortcut icon" href="../_static/bigomics-logo-white-48px.png"/>
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Data View" href="mod1_dataview.html" />
    <link rel="prev" title="Outline" href="outline.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            Omics Playground
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../overview.html">Overview</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Basic Data Preparation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/dataprep.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/samples.html">Samples file</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/counts.html">Counts file</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/contrasts.html">Contrasts file (optional)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/uploadopg.html">Uploading your data in Omics Playground</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Advanced Data Preparation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/dataprep_example.html">Preparing input files from raw data formats</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/geneset.html">Loading custom geneset files</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/dataprocess.html">Data processing inside Omics Playground</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Analysis Modules</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="outline.html">Outline</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Home Panel</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#selecting-the-dataset">Selecting the dataset</a></li>
<li class="toctree-l2"><a class="reference internal" href="#upload-data-upload">Upload data (Upload)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#upload-data-contrasts">Upload data (Contrasts)</a></li>
<li class="toctree-l2"><a class="reference internal" href="#upload-data-compute">Upload data (Compute)</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="mod1_dataview.html">Data View</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod2_clustering.html">Clustering</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod3_DEGs.html">Expression</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod4_DEGSs.html">Enrichment</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod7_Signature.html">Signature</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod9_CellProfiling.html">Cell Profiling</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Examples</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../examples/examples.html">Reanalyzing Public Datasets</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Methods</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../methods.html">Methods</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">FAQ</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../faq.html">Frequently Asked Questions (FAQ)</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">About</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../info.html">Contact</a></li>
<li class="toctree-l1"><a class="reference internal" href="../info.html#citation">Citation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../info.html#license">License</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Omics Playground</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Home Panel</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/modules/mod0_home.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="home-panel">
<span id="home"></span><h1>Home Panel<a class="headerlink" href="#home-panel" title="Permalink to this headline">¶</a></h1>
<section id="selecting-the-dataset">
<h2>Selecting the dataset<a class="headerlink" href="#selecting-the-dataset" title="Permalink to this headline">¶</a></h2>
<p>The platform starts running from the <strong>Home</strong> panel. This panel shows
the available datasets within the platform. The table reports a brief
description as well as the total number of samples, genes, gene sets
(or pathways), corresponding phenotypes and the collection date.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc1.2.png"><img alt="../_images/psc1.2.png" src="../_images/psc1.2.png" style="width: 100%;" /></a>
</figure>
<p>Users can select a dataset in the table. The <code class="docutils literal notranslate"><span class="pre">Dataset</span> <span class="pre">info</span></code> shows
the information of the dataset of interest and users can load the data
by clicking the <code class="docutils literal notranslate"><span class="pre">Load</span></code> button. There are three additional icons to
dwonload pgx files, download csv files and delete an entry.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc1.0.png"><img alt="../_images/psc1.0.png" src="../_images/psc1.0.png" style="width: 30%;" /></a>
</figure>
</section>
<section id="upload-data-upload">
<h2>Upload data (Upload)<a class="headerlink" href="#upload-data-upload" title="Permalink to this headline">¶</a></h2>
<p>Under the <strong>Upload</strong> subpanel of the <strong>Upload data</strong> panel users can upload their transcriptomics
and proteomics data to the platform. The platform requires three data
files as listed below: a data file containing counts/expression
(counts.csv), a sample information file (samples.csv) and an optional
file specifying the statistical comparisons as contrasts (contrasts.csv).
The file format must be comma-separated-values (CSV) text.
Be sure the gene names match for all files. On the left side of the panel,
users need to provide a unique name and brief description for the dataset
while uploading.</p>
<dl class="field-list simple">
<dt class="field-odd"><strong>counts.csv</strong></dt>
<dd class="field-odd"><p>Count/expression file with gene on rows, samples as columns.</p>
</dd>
<dt class="field-even"><strong>samples.csv</strong></dt>
<dd class="field-even"><p>Samples file with samples on rows, phenotypes as columns.</p>
</dd>
<dt class="field-odd"><strong>contrasts.csv</strong></dt>
<dd class="field-odd"><p>Contrast file with conditions on rows, contrasts as columns.</p>
</dd>
</dl>
<p>Once uploaded, the platform generates three separate plots showing some stastics
and the structure of your input files.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc1.3.png"><img alt="../_images/psc1.3.png" src="../_images/psc1.3.png" style="width: 100%;" /></a>
</figure>
<p>The page also contains a <em>batch correction</em> option that allows users to perform batch correction
on the data. Selecting it will open a new panel from which users can select the level of
batch correction (low, medium or strong). Under <em>Advanced</em> users can fine-tune
the process. This feature is still in beta and only recommended to users familiar with the
batch correction precedure and parameters. More information can be found under <a class="reference internal" href="../methods.html#methods"><span class="std std-ref">Methods</span></a>.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc1.3.a.png"><img alt="../_images/psc1.3.a.png" src="../_images/psc1.3.a.png" style="width: 100%;" /></a>
</figure>
</section>
<section id="upload-data-contrasts">
<h2>Upload data (Contrasts)<a class="headerlink" href="#upload-data-contrasts" title="Permalink to this headline">¶</a></h2>
<p>Users can eschew adding a contrast file and instead generate contrasts via
the <strong>Contrasts</strong> subpanel of the <strong>Upload data</strong> panel.
Available phenotypes will appear in the “Phenotypes” box as a scrolldown menu.
Selecting a phenotype will show the available conditions in the “Conditions” box.
Users can then drag individual groups in the “Main group” or “Control Group” boxes.
It is also possible to select individual samples from the “Conditions” box and phenotypes
can be furthermore combined in the “Phenotypes” box. The platform automatically generates
a contrast name in the “Comparison name” box that users can manually edit. Clicking the
“add comparison” button will add the selected contrast in the contrast table below.
Users can also task the platform to generate comparisons automatically by selecting a
phenotype form the “Strata” box and clicking on “add auto-contrasts”. However, we recommend
caution when using this function as it is error-prone.
The panel also produces a PCA/tSNE plot that users can consult as a guide for generating
comparisons.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc1.4.png"><img alt="../_images/psc1.4.png" src="../_images/psc1.4.png" style="width: 100%;" /></a>
</figure>
</section>
<section id="upload-data-compute">
<h2>Upload data (Compute)<a class="headerlink" href="#upload-data-compute" title="Permalink to this headline">¶</a></h2>
<p>The <strong>Compute</strong> subpanel is where users can start the computation of their data.
They need to provide a unique name for the dataset, indicate the type of data and
provide a short description. Beginners can then click on the “Compute!” button and
let the platform run the calculations. Advanced users can click on “Advanced” to
access several customisation options. In particular they can de-select feature
filtering options, select different gene tests and enrichment methods combinations,
include or exclude analysis types and select developer-specific options.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc1.5.png"><img alt="../_images/psc1.5.png" src="../_images/psc1.5.png" style="width: 100%;" /></a>
</figure>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="outline.html" class="btn btn-neutral float-left" title="Outline" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="mod1_dataview.html" class="btn btn-neutral float-right" title="Data View" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright .</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>