<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Expression &mdash; Omics Playground  documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
    <link rel="shortcut icon" href="../_static/bigomics-logo-white-48px.png"/>
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Enrichment" href="mod4_DEGSs.html" />
    <link rel="prev" title="Clustering" href="mod2_clustering.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            Omics Playground
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../overview.html">Overview</a></li>
<li class="toctree-l1"><a class="reference internal" href="../install.html">Installation</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Data Preparation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/dataprep.html">Data import and precomputation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/dataprep_example.html">Data preparation examples</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/samples.html">Manual to prepare the <cite>samples.csv</cite> file</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Analysis Modules</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="outline.html">Outline</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod0_home.html">Home Panel</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod1_dataview.html">Data View</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod2_clustering.html">Clustering</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Expression</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#differential-expression">Differential expression</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#input-panel">Input panel</a></li>
<li class="toctree-l3"><a class="reference internal" href="#table">Table</a></li>
<li class="toctree-l3"><a class="reference internal" href="#foldchange-all">Foldchange (all)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#fdr-table">FDR table</a></li>
<li class="toctree-l3"><a class="reference internal" href="#plot">Plot</a></li>
<li class="toctree-l3"><a class="reference internal" href="#top-genes">Top genes</a></li>
<li class="toctree-l3"><a class="reference internal" href="#volcano-all">Volcano (all)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#volcano-methods">Volcano (methods)</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#correlation-analysis">Correlation analysis</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id2">Input panel</a></li>
<li class="toctree-l3"><a class="reference internal" href="#correllation">Correllation</a></li>
<li class="toctree-l3"><a class="reference internal" href="#functional">Functional</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="mod4_DEGSs.html">Enrichment</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod7_Signature.html">Signature</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod9_CellProfiling.html">Cell Profiling</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Examples</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../examples/examples.html">Reanalyzing Public Datasets</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Methods</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../methods.html">Methods</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">FAQ</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../faq.html">Frequently Asked Questions (FAQ)</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">About</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../info.html">Contact</a></li>
<li class="toctree-l1"><a class="reference internal" href="../info.html#citation">Citation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../info.html#license">License</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Omics Playground</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Expression</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/modules/mod3_DEGs.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="expression">
<span id="id1"></span><h1>Expression<a class="headerlink" href="#expression" title="Permalink to this headline">¶</a></h1>
<p>The <strong>Expression</strong> module is divided into two main panels. The <strong>Differential Expression</strong>
panel compares expression between two conditions (i.e. tumor versus control),
which is one of the fundamental analysis in the transcriptomics data analytics workflow.
For each comparison of two conditions (also called a ‘contrast’), the analysis identifies
which genes are significantly downregulated or overexpressed in one of the groups.</p>
<p>The <strong>Correlation Analysis</strong> panel computes the correlation between genes and finds
coregulated modules.</p>
<section id="differential-expression">
<h2>Differential expression<a class="headerlink" href="#differential-expression" title="Permalink to this headline">¶</a></h2>
<section id="input-panel">
<h3>Input panel<a class="headerlink" href="#input-panel" title="Permalink to this headline">¶</a></h3>
<p>The input panel on the left contains a link to a tutorial for relevant, as well as some settings
for the analysis. Users can start the differntial expression (DE) analysis by selecting a contrast
of their interest in the <code class="docutils literal notranslate"><span class="pre">Contrast</span></code> setting and specifying a relevent gene family in the
<code class="docutils literal notranslate"><span class="pre">Gene</span> <span class="pre">family</span></code> setting. It is possible to set the false discovery rate (FDR) and the
logarithmic fold change (logFC) thresholds under the <code class="docutils literal notranslate"><span class="pre">FDR</span></code> and <code class="docutils literal notranslate"><span class="pre">logFC</span> <span class="pre">threshold</span></code> settings, respectively.
Under “Options”, users can select to display all genes in the table (not only significant genes)
and also select different combinations of statistical methods for the analysis.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc4.0.png"><img alt="../_images/psc4.0.png" src="../_images/psc4.0.png" style="width: 30%;" /></a>
</figure>
</section>
<section id="table">
<h3>Table<a class="headerlink" href="#table" title="Permalink to this headline">¶</a></h3>
<p>Table <code class="docutils literal notranslate"><span class="pre">I</span></code> shows the results of the statistical tests slected in the
<code class="docutils literal notranslate"><span class="pre">Statistical</span> <span class="pre">methods</span></code>. By default, this table reports
the meta (combined) results of
<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4302049/">DESeq2</a> (Wald),
<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/19910308">edgeR</a> (QLF), and
<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/25605792">limma</a> (trend) only.
Users can filter top N = {10} differently expressed genes in the table by
clicking the <code class="docutils literal notranslate"><span class="pre">top</span> <span class="pre">10</span> <span class="pre">genes</span></code>  and also show the q-values from individual statistical methods
by ticking <code class="docutils literal notranslate"><span class="pre">show</span> <span class="pre">individual</span> <span class="pre">q-values</span></code> from the table <em>Settings</em>.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc4.1.0.png"><img alt="../_images/psc4.1.0.png" src="../_images/psc4.1.0.png" style="width: 30%;" /></a>
</figure>
<p>For a selected comparison under the <code class="docutils literal notranslate"><span class="pre">Contrast</span></code> setting, the results of the selected
methods are combined and reported in Table <code class="docutils literal notranslate"><span class="pre">I</span></code>, where <code class="docutils literal notranslate"><span class="pre">meta.q</span></code> for a gene
represents the highest <code class="docutils literal notranslate"><span class="pre">q</span></code> value among the methods and the number of stars for
a gene indicate how many methods identified significant <code class="docutils literal notranslate"><span class="pre">q</span></code> values (<code class="docutils literal notranslate"><span class="pre">q</span> <span class="pre">&lt;</span> <span class="pre">0.05</span></code>).
The table is interactive (scrollable, clickable); users can sort genes by <code class="docutils literal notranslate"><span class="pre">logFC</span></code>,
<code class="docutils literal notranslate"><span class="pre">meta.q</span></code>, or average expression in either conditions.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc4.1.png"><img alt="../_images/psc4.1.png" src="../_images/psc4.1.png" style="width: 100%;" /></a>
</figure>
<p>By clicking on a gene in the Table <code class="docutils literal notranslate"><span class="pre">I</span></code> (highlighted in purple in the figure),
it is possible to see the correlation and enrichment value of gene sets that
contain the gene in Table <code class="docutils literal notranslate"><span class="pre">II</span></code>.</p>
</section>
<section id="foldchange-all">
<h3>Foldchange (all)<a class="headerlink" href="#foldchange-all" title="Permalink to this headline">¶</a></h3>
<p>The <strong>Foldchange (all)</strong> tab reports the gene fold changes for all contrasts in the selected dataset.
The column <code class="docutils literal notranslate"><span class="pre">fc.var</span></code> corresponds to the variance of the fold-change across all contrasts.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc4.1.1.png"><img alt="../_images/psc4.1.1.png" src="../_images/psc4.1.1.png" style="width: 100%;" /></a>
</figure>
</section>
<section id="fdr-table">
<h3>FDR table<a class="headerlink" href="#fdr-table" title="Permalink to this headline">¶</a></h3>
<p>The <strong>FDR table</strong> tab reports the number of significant genes at different FDR thresholds for
all contrasts and methods within the dataset. This enables to quickly see which
methods are more sensitive. The left part of the table (in blue) correspond
to the number of significant down-regulated genes, the right part (in red)
correspond to the number of significant overexpressed genes.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc4.1.2.png"><img alt="../_images/psc4.1.2.png" src="../_images/psc4.1.2.png" style="width: 100%;" /></a>
</figure>
</section>
<section id="plot">
<h3>Plot<a class="headerlink" href="#plot" title="Permalink to this headline">¶</a></h3>
<p>The <strong>Plot</strong> panel shows the following plots:</p>
<dl class="field-list simple">
<dt class="field-odd"><strong>a</strong></dt>
<dd class="field-odd"><p>An interactive volcano plot for the chosen contrast. Unless a specific gene is selected
from the differential expression analysis table, all significant genes are highlighted in blue.</p>
</dd>
<dt class="field-even"><strong>b</strong></dt>
<dd class="field-even"><p>An interactive MA plot for the chosen contrast. Unless a specific gene is selected
from the differential expression analysis table, all significant genes are highlighted in blue.</p>
</dd>
<dt class="field-odd"><strong>c</strong></dt>
<dd class="field-odd"><p>Differential expression boxplot for a gene that is selected from the
differential expression analysis table. Users can customise the plot via the settings icon on top
to ungroup samples and change the scale to counts per million (CPM).</p>
</dd>
<dt class="field-even"><strong>d</strong></dt>
<dd class="field-even"><p>Fold change summary barplot across all contrasts for a gene that is selected
from the differential expression analysis table.</p>
</dd>
</dl>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc4.1.3.png"><img alt="../_images/psc4.1.3.png" src="../_images/psc4.1.3.png" style="width: 100%;" /></a>
</figure>
</section>
<section id="top-genes">
<h3>Top genes<a class="headerlink" href="#top-genes" title="Permalink to this headline">¶</a></h3>
<p>The <strong>Top genes</strong> section shows the average expression plots across the samples for the top differentially
(both positively and negatively) expressed genes for the selected comparison from the <code class="docutils literal notranslate"><span class="pre">Contrast</span></code> setting.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc4.2.png"><img alt="../_images/psc4.2.png" src="../_images/psc4.2.png" style="width: 100%;" /></a>
</figure>
<p>The plot can be customised via the settings to remove the log scale, ungroup samples and show only samples
included in the selected contrast.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc4.2.0.png"><img alt="../_images/psc4.2.0.png" src="../_images/psc4.2.0.png" style="width: 30%;" /></a>
</figure>
</section>
<section id="volcano-all">
<h3>Volcano (all)<a class="headerlink" href="#volcano-all" title="Permalink to this headline">¶</a></h3>
<p>Under the <strong>Volcano (all)</strong> tab, the platform simultaneously displays multiple volcano plots
for genes across all contrasts. By comparing multiple volcano plots,
the user can immediately see which comparison is statistically weak or strong.
Experimental contrasts with better statistical significance will show
volcano plots with ‘higher’ wings.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc4.3.png"><img alt="../_images/psc4.3.png" src="../_images/psc4.3.png" style="width: 100%;" /></a>
</figure>
</section>
<section id="volcano-methods">
<h3>Volcano (methods)<a class="headerlink" href="#volcano-methods" title="Permalink to this headline">¶</a></h3>
<p>Under the <strong>Volcano (methods)</strong> tab, the platform displays the volcano plots provided by
multiple differential expression calculation methods for the selected contrast.
Methods showing better statistical significance will show volcano
plots with ‘higher’ wings.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc4.4.png"><img alt="../_images/psc4.4.png" src="../_images/psc4.4.png" style="width: 100%;" /></a>
</figure>
</section>
</section>
<section id="correlation-analysis">
<h2>Correlation analysis<a class="headerlink" href="#correlation-analysis" title="Permalink to this headline">¶</a></h2>
<section id="id2">
<h3>Input panel<a class="headerlink" href="#id2" title="Permalink to this headline">¶</a></h3>
<p>Users can find more information by clicking <code class="docutils literal notranslate"><span class="pre">Info</span></code>. The panel contains the main settings for the analysis.
The analysis can be started by selecting a gene of interest from the <code class="docutils literal notranslate"><span class="pre">Gene</span></code> settings.
Under <em>Options</em>, users can filter for a specific gene family by using the <code class="docutils literal notranslate"><span class="pre">Gene</span> <span class="pre">family</span></code> setting.
By using the <code class="docutils literal notranslate"><span class="pre">Filter</span> <span class="pre">samples</span></code> setting they con focus only on specific sample groups.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc4.5.0.png"><img alt="../_images/psc4.5.0.png" src="../_images/psc4.5.0.png" style="width: 30%;" /></a>
</figure>
</section>
<section id="correllation">
<h3>Correllation<a class="headerlink" href="#correllation" title="Permalink to this headline">¶</a></h3>
<p>Under the <strong>Correlation</strong> tab, the platform displays three different outputs:</p>
<dl class="field-list simple">
<dt class="field-odd"><strong>a</strong></dt>
<dd class="field-odd"><p>A plot displaying the highest correlated genes in respect to the selected gene.
The plot can be customised via the settings icon on top to include no partial correlection,
a fast partial correlation calculation or a calculation with all methods.</p>
</dd>
</dl>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc4.5.1.png"><img alt="../_images/psc4.5.1.png" src="../_images/psc4.5.1.png" style="width: 30%;" /></a>
</figure>
<dl class="field-list simple">
<dt class="field-odd"><strong>b</strong></dt>
<dd class="field-odd"><p>A correlation graph centered on selected gene with top most correlated features.
The plot can be customised via the settings icon on top. From it users can adjust the rho value
via a sliding bar and change the graph layout via <code class="docutils literal notranslate"><span class="pre">Layout</span></code>.</p>
</dd>
</dl>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc4.5.2.png"><img alt="../_images/psc4.5.2.png" src="../_images/psc4.5.2.png" style="width: 30%;" /></a>
</figure>
<dl class="field-list simple">
<dt class="field-odd"><strong>c</strong></dt>
<dd class="field-odd"><p>Pairwise scatter plots for the expression values of the gene pairs across the samples.</p>
</dd>
</dl>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc4.5.png"><img alt="../_images/psc4.5.png" src="../_images/psc4.5.png" style="width: 100%;" /></a>
</figure>
</section>
<section id="functional">
<h3>Functional<a class="headerlink" href="#functional" title="Permalink to this headline">¶</a></h3>
<p>Four outputs are generated under the <strong>Functional</strong> tab:</p>
<dl class="field-list simple">
<dt class="field-odd"><strong>a</strong></dt>
<dd class="field-odd"><p>A series of gene set enrichment analysis (GSEA) plots using the correlation as rank metric.
The black bars denote the genes in the gene set and their position in the sorted rank metric.</p>
</dd>
<dt class="field-even"><strong>b</strong></dt>
<dd class="field-even"><p>A table containing the statistical results from the GSEA. The table displays
normalised enrichment scores (NES), adjusted and non-adjusted p-values,size of the gene set
and a link to gene set information.</p>
</dd>
<dt class="field-odd"><strong>c</strong></dt>
<dd class="field-odd"><p>A plot of the frequency of leading edge genes in top correlated genesets.</p>
</dd>
<dt class="field-even"><strong>d</strong></dt>
<dd class="field-even"><p>A table of the leading edge genes and rank statistics (rho) of the selected geneset.</p>
</dd>
</dl>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc4.6.png"><img alt="../_images/psc4.6.png" src="../_images/psc4.6.png" style="width: 100%;" /></a>
</figure>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="mod2_clustering.html" class="btn btn-neutral float-left" title="Clustering" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="mod4_DEGSs.html" class="btn btn-neutral float-right" title="Enrichment" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright .</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>