<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Enrichment &mdash; Omics Playground  documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
    <link rel="shortcut icon" href="../_static/bigomics-logo-white-48px.png"/>
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Signature" href="mod7_Signature.html" />
    <link rel="prev" title="Expression" href="mod3_DEGs.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            Omics Playground
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../overview.html">Overview</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Basic Data Preparation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/dataprep.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/samples.html">Samples file</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/counts.html">Counts file</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/contrasts.html">Contrasts file (optional)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/uploadopg.html">Uploading your data in Omics Playground</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Advanced Data Preparation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/dataprep_example.html">Preparing input files from raw data formats</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/geneset.html">Loading custom geneset files</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/dataprocess.html">Data processing inside Omics Playground</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Analysis Modules</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="outline.html">Outline</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod0_home.html">Home Panel</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod1_dataview.html">Data View</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod2_clustering.html">Clustering</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod3_DEGs.html">Expression</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Enrichment</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#geneset-enrichment">Geneset enrichment</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#input-panel">Input panel</a></li>
<li class="toctree-l3"><a class="reference internal" href="#table">Table</a></li>
<li class="toctree-l3"><a class="reference internal" href="#top-enriched">Top enriched</a></li>
<li class="toctree-l3"><a class="reference internal" href="#plots">Plots</a></li>
<li class="toctree-l3"><a class="reference internal" href="#compare">Compare</a></li>
<li class="toctree-l3"><a class="reference internal" href="#volcano-all">Volcano (all)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#volcano-methods">Volcano (methods)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#genemap">GeneMap</a></li>
<li class="toctree-l3"><a class="reference internal" href="#foldchange-all">Foldchange (all)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#fdr-table">FDR table</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#pathway-analysis">Pathway analysis</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id2">Input panel</a></li>
<li class="toctree-l3"><a class="reference internal" href="#kegg-pathways">KEGG pathways</a></li>
<li class="toctree-l3"><a class="reference internal" href="#go-graph">GO graph</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#word-cloud">Word Cloud</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id3">Input panel</a></li>
<li class="toctree-l3"><a class="reference internal" href="#main-panel">Main panel</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#drug-connectivity">Drug connectivity</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id4">Input panel</a></li>
<li class="toctree-l3"><a class="reference internal" href="#id5">Main panel</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="mod7_Signature.html">Signature</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod9_CellProfiling.html">Cell Profiling</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Examples</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../examples/examples.html">Reanalyzing Public Datasets</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Methods</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../methods.html">Methods</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">FAQ</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../faq.html">Frequently Asked Questions (FAQ)</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">About</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../info.html">Contact</a></li>
<li class="toctree-l1"><a class="reference internal" href="../info.html#citation">Citation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../info.html#license">License</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Omics Playground</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Enrichment</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/modules/mod4_DEGSs.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="enrichment">
<span id="id1"></span><h1>Enrichment<a class="headerlink" href="#enrichment" title="Permalink to this headline">¶</a></h1>
<p>This panel is subdivided into four subpanels: <strong>Geneset enrichment</strong>,
<strong>Pathway analysis</strong>, <strong>Word cloud</strong> and <strong>Drug connectivity</strong>.</p>
<section id="geneset-enrichment">
<h2>Geneset enrichment<a class="headerlink" href="#geneset-enrichment" title="Permalink to this headline">¶</a></h2>
<p>Similar to the differential gene expression analysis, users can perform differential
expression analysis on a geneset level in this page, which is also referred as
gene set enrichment (GSE) analysis. The platform has more than 50.000 genesets
(or pathways) in total that are divided into 30 geneset collections such
<a class="reference external" href="https://www.cell.com/abstract/S0092-8674(11)00127-9">Hallmark</a>,
<a class="reference external" href="http://software.broadinstitute.org/gsea/msigdb/index.jsp">MSigDB</a>,
<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC102409/">KEGG</a>,
and Gene Ontology (<a class="reference external" href="http://geneontology.org/">GO</a>).
Users have to specify which comparison they want to visually analyze
employing a certain geneset collection.</p>
<p>The enrichment analyses results from the statistical methods are
displayed in <strong>Table</strong> panel. In the <strong>Top enriched</strong> panel, the top
differentially enriched geneses (pathways) are displayed. In the <strong>Plots</strong> panel,
a volcano plot of genes contained in the selected geneset and a barplot of
expressions per sample group are displayed. In the <strong>Compare</strong> panel, users can
compare the differential expression status of that geneset for all other
comparisons. Volcano plots of genesets for all comparisons are
displayed under the <strong>Volcano (all)</strong> tab. This allows users to have an overall
picture across comparisons at the same time. To compare the different statistical methods,
the <strong>Volcano (methods)</strong> panel shows volcano plots of all methods. The <strong>GeneMap</strong> panel
displays the co-activation heatmap of top N = {25} enriched gene sets and their common genes.
The <strong>Foldchange (all)</strong> tab reports the gene fold changes for all contrasts in the selected dataset.
The column <code class="docutils literal notranslate"><span class="pre">fc.var</span></code> corresponds to the variance of the fold-change across all contrasts.
The <strong>FDR table</strong> panel reports the number of significant gene sets at different FDR thresholds for all contrasts.</p>
<section id="input-panel">
<h3>Input panel<a class="headerlink" href="#input-panel" title="Permalink to this headline">¶</a></h3>
<p>Users can find a tutorial about the module under <code class="docutils literal notranslate"><span class="pre">Tutorial</span></code> in the input panel.
The enrichment analysis can be started by selecting a contrats of interest in
the <code class="docutils literal notranslate"><span class="pre">Contrast</span></code> and specifying a relevent gene set family in the <code class="docutils literal notranslate"><span class="pre">Gene</span> <span class="pre">set</span> <span class="pre">collection</span></code>.
It is possible to set the false discovery rate (FDR) and the logarithmic fold change
(logFC) thresholds under the <code class="docutils literal notranslate"><span class="pre">FDR</span></code> and <code class="docutils literal notranslate"><span class="pre">logFC</span> <span class="pre">threshold</span></code> settings, respectively.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc5.0.png"><img alt="../_images/psc5.0.png" src="../_images/psc5.0.png" style="width: 30%;" /></a>
</figure>
<p>We allow users to show all gene sets in table <code class="docutils literal notranslate"><span class="pre">I</span></code> , rather than just the significant ones, under the <em>Options</em> menu.
Users can also select statistical methods for the enrichment analysis from the same menu.
To ensure the statistical reliability, the platform performs enrichment analyses using multiple methods, including
<a class="reference external" href="https://en.wikipedia.org/wiki/Spearman%27s_rank_correlation_coefficient">Spearman rank correlation</a>,
<a class="reference external" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-7">GSVA</a>,
<a class="reference external" href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-7">ssGSEA</a>,
<a class="reference external" href="https://www.jstor.org/stable/2340521?seq=1#metadata_info_tab_contents">Fisher’s exact test</a>,
<a class="reference external" href="http://software.broadinstitute.org/gsea/index.jsp">GSEA</a>,
<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3458527/">camera</a> and
<a class="reference external" href="https://academic.oup.com/bioinformatics/article/26/17/2176/200022">fry</a>.</p>
</section>
<section id="table">
<h3>Table<a class="headerlink" href="#table" title="Permalink to this headline">¶</a></h3>
<p>Table <code class="docutils literal notranslate"><span class="pre">I</span></code> shows the enrichment analysis results from the selected statistical methods.
WIth default settings, this table reports the meta (combined) results of
<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3458527/">camera</a>,
<a class="reference external" href="http://software.broadinstitute.org/gsea/index.jsp">fgsea</a>, and
<a class="reference external" href="https://en.wikipedia.org/wiki/Spearman%27s_rank_correlation_coefficient">Spearman rank correlation</a> only.
Users can filter top N = {10} differently enriched gene sets in the table by
clicking the <code class="docutils literal notranslate"><span class="pre">top</span> <span class="pre">10</span> <span class="pre">gene</span> <span class="pre">sets</span></code> from the table <em>Settings</em>. They can also select to display
individual q-values.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc5.1.0.png"><img alt="../_images/psc5.1.0.png" src="../_images/psc5.1.0.png" style="width: 30%;" /></a>
</figure>
<p>The combined enrichment analysis results from the methods are displayed in Table <code class="docutils literal notranslate"><span class="pre">I</span></code>,
where for each geneset the <code class="docutils literal notranslate"><span class="pre">meta.q</span></code> corresponds to the highest <code class="docutils literal notranslate"><span class="pre">q</span></code> value provided
by the methods and the number of <code class="docutils literal notranslate"><span class="pre">stars</span></code> indicate how many methods identified
the geneset as significant (<code class="docutils literal notranslate"><span class="pre">q</span> <span class="pre">&lt;</span> <span class="pre">0.05</span></code>). <code class="docutils literal notranslate"><span class="pre">AveExpr0</span></code> and <code class="docutils literal notranslate"><span class="pre">AveEprx1</span></code> refer to the average gene set expression
in group 0 and group 1 of the selected pairwise comparison respectively.
The table is interactive; users can sort it by <code class="docutils literal notranslate"><span class="pre">logFC</span></code>, <code class="docutils literal notranslate"><span class="pre">meta.q</span></code>, <code class="docutils literal notranslate"><span class="pre">AveExpr0</span></code>, <code class="docutils literal notranslate"><span class="pre">AveEprx1</span></code> and <code class="docutils literal notranslate"><span class="pre">stars</span></code>.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc5.1.png"><img alt="../_images/psc5.1.png" src="../_images/psc5.1.png" style="width: 100%;" /></a>
</figure>
<p>By clicking on a gene set in Table <code class="docutils literal notranslate"><span class="pre">I</span></code>, it is possible to see the gene list of
that gene set in Table <code class="docutils literal notranslate"><span class="pre">II</span></code>. It also reports the fold-change, statistics and
correlation of the genes in the selected gene set. By clicking on a gene in
Table <code class="docutils literal notranslate"><span class="pre">II</span></code>, under the <strong>Plots</strong> panel, users can check its correlation to the gene set
in the <code class="docutils literal notranslate"><span class="pre">Enrichment</span> <span class="pre">vs.</span> <span class="pre">expression</span></code> correlation scatter plot.</p>
</section>
<section id="top-enriched">
<h3>Top enriched<a class="headerlink" href="#top-enriched" title="Permalink to this headline">¶</a></h3>
<p>The <strong>Top enriched</strong> panel shows two plots:</p>
<dl class="field-list simple">
<dt class="field-odd"><strong>a</strong></dt>
<dd class="field-odd"><p>The enrichment plots of the top differentially
enriched gene sets (up and down) for the selected contrast. Black vertical bars
indicate the rank of genes in the gene set in the sorted list metric.
The green curve corresponds to the enrichment score (ES). The more the green
ES curve is shifted to the upper left of the graph, the more the gene set
is enriched in the first group. Conversely, a shift of the ES curve to the
lower right, corresponds to more enrichment in the second group.</p>
</dd>
<dt class="field-even"><strong>b</strong></dt>
<dd class="field-even"><p>A gene frequency plot shows the number of times a gene is present in the top-N genesets
sorted by frequency. Genes that are frequently shared among the top enriched gene sets
may suggest driver genes.</p>
</dd>
</dl>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc5.2.png"><img alt="../_images/psc5.2.png" src="../_images/psc5.2.png" style="width: 100%;" /></a>
</figure>
<p>The gene frequency plot can be customised via the <em>Settings</em>. Users can change the nuber of
top gene sets to be included and deselect weighting by fold change (FC) and gene set size.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc5.2.0.png"><img alt="../_images/psc5.2.0.png" src="../_images/psc5.2.0.png" style="width: 30%;" /></a>
</figure>
</section>
<section id="plots">
<h3>Plots<a class="headerlink" href="#plots" title="Permalink to this headline">¶</a></h3>
<p>The <strong>Plots</strong> panel provide plots associated with the gene set
(selected in Table <code class="docutils literal notranslate"><span class="pre">I</span></code>) and gene (selected in Table <code class="docutils literal notranslate"><span class="pre">II</span></code>) under <code class="docutils literal notranslate"><span class="pre">Table</span></code>.</p>
<dl class="field-list simple">
<dt class="field-odd"><strong>a</strong></dt>
<dd class="field-odd"><p>Enrichment plot associated with the gene set selected from Table <code class="docutils literal notranslate"><span class="pre">I</span></code>.</p>
</dd>
<dt class="field-even"><strong>b</strong></dt>
<dd class="field-even"><p>Volcano-plot of genes showing the significance versus the fold-change on the y and x axes,
respectively. Genes in the selected gene set are highlighted in blue.</p>
</dd>
<dt class="field-odd"><strong>c</strong></dt>
<dd class="field-odd"><p>Barplot of the gene expression of the gene. Samples in the barplot
can be ungrouped in the plot <em>Settings</em>.</p>
</dd>
<dt class="field-even"><strong>d</strong></dt>
<dd class="field-even"><p>Scatter plot of the enrichment versus the expression of the selected
geneset and gene, on the y and x axes, respectively.</p>
</dd>
</dl>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc5.3.png"><img alt="../_images/psc5.3.png" src="../_images/psc5.3.png" style="width: 100%;" /></a>
</figure>
</section>
<section id="compare">
<h3>Compare<a class="headerlink" href="#compare" title="Permalink to this headline">¶</a></h3>
<p>Under the <strong>Compare</strong> panel, users can fin enrichment plots for the selected gene
set (in Table <code class="docutils literal notranslate"><span class="pre">I</span></code>) across multiple contrasts. The figure allows to quickly
compare the enrichment of a certain gene set across all other comparisons.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc5.4.png"><img alt="../_images/psc5.4.png" src="../_images/psc5.4.png" style="width: 100%;" /></a>
</figure>
</section>
<section id="volcano-all">
<h3>Volcano (all)<a class="headerlink" href="#volcano-all" title="Permalink to this headline">¶</a></h3>
<p>The <strong>Volcano (all)</strong> panel simultaneously displays volcano plots of gene sets
enrichment across all contrasts, showing the enrichment score versus significance
on the x and y axes, respectively. This provides users an overview of the
statistics across all comparisons. By comparing multiple volcano plots, the user
can immediately see which comparison is statistically weak or strong.
Experimental contrasts showing better statistical significance will show volcano
plots with ‘higher’ wings.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc5.5.png"><img alt="../_images/psc5.5.png" src="../_images/psc5.5.png" style="width: 100%;" /></a>
</figure>
</section>
<section id="volcano-methods">
<h3>Volcano (methods)<a class="headerlink" href="#volcano-methods" title="Permalink to this headline">¶</a></h3>
<p>Under the <strong>Volcano (methods)</strong> panel, users can see the simultaneous
visualisation of volcano plots of gene sets for different enrichment methods.
This provides users an quick overview of the sensitivity of the statistical
methods at once. Methods showing better statistical significance will show
volcano plots with ‘higher’ wings.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc5.6.png"><img alt="../_images/psc5.6.png" src="../_images/psc5.6.png" style="width: 100%;" /></a>
</figure>
</section>
<section id="genemap">
<h3>GeneMap<a class="headerlink" href="#genemap" title="Permalink to this headline">¶</a></h3>
<p>Under the <strong>GeneMap</strong> panel, users can view a co-activation heatmap of the top N = {25}
enriched gene sets and their common genes.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc5.9.png"><img alt="../_images/psc5.9.png" src="../_images/psc5.9.png" style="width: 100%;" /></a>
</figure>
</section>
<section id="foldchange-all">
<h3>Foldchange (all)<a class="headerlink" href="#foldchange-all" title="Permalink to this headline">¶</a></h3>
<p>The <strong>Foldchange (all)</strong> provides the enrichment analysis of gene sets
across all contrasts. The column <code class="docutils literal notranslate"><span class="pre">fc.var</span></code> corresponds to the variance of
the fold-change across all contrasts.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc5.7.png"><img alt="../_images/psc5.7.png" src="../_images/psc5.7.png" style="width: 100%;" /></a>
</figure>
</section>
<section id="fdr-table">
<h3>FDR table<a class="headerlink" href="#fdr-table" title="Permalink to this headline">¶</a></h3>
<p>The <strong>FDR table</strong> panel reports the number of significant gene sets at different
FDR thresholds for all contrasts and methods. Using the table the user can
determine which statistical methods perform better for a particular contrast.
The left part of the table (in blue) correspond
to the number of significant down-regulated gene sets, the right part (in red)
correspond to the number of significant overexpressed gene sets.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc5.8.png"><img alt="../_images/psc5.8.png" src="../_images/psc5.8.png" style="width: 100%;" /></a>
</figure>
</section>
</section>
<section id="pathway-analysis">
<h2>Pathway analysis<a class="headerlink" href="#pathway-analysis" title="Permalink to this headline">¶</a></h2>
<p>This module performs specialized pathway and enrichment analysis
providing higher level functional and visual analysis</p>
<p>The <strong>KEGG pathways</strong> panel maps the differential fold-changes onto
the <a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC102409/">KEGG</a>
pathway maps. Under the <strong>GO</strong> panel, a graph-based enrichment
analysis is done using the Gene Ontology (<a class="reference external" href="http://geneontology.org/">GO</a>) graph structure.</p>
<section id="id2">
<h3>Input panel<a class="headerlink" href="#id2" title="Permalink to this headline">¶</a></h3>
<p>A tutorial about the module can be found under <code class="docutils literal notranslate"><span class="pre">Youtube</span></code> in the input panel.
Users can specify the contrast of their interest
in the <code class="docutils literal notranslate"><span class="pre">Contrast</span></code> settings. Under the main <em>Options</em>, users can
select <code class="docutils literal notranslate"><span class="pre">filter</span> <span class="pre">significant</span> <span class="pre">(tables)</span></code>  to keep only significant entries in the table.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc6.0.png"><img alt="../_images/psc6.0.png" src="../_images/psc6.0.png" style="width: 30%;" /></a>
</figure>
</section>
<section id="kegg-pathways">
<h3>KEGG pathways<a class="headerlink" href="#kegg-pathways" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC102409/">KEGG</a> is a collection
of manually curated pathways representing the current knowledge of molecular
interactions, reactions and relation networks as pathway maps. In the
<strong>KEGG pathway</strong> panel, each pathway is scored for the selected contrast profile
and reported in the table. A unique feature of the platform is that it provides
an activation-heatmap comparing the activation levels of pathways across multiple
contrast profiles. This facilitates to quickly see and detect the similarities
between profiles in certain pathways. More detailed explaination of each output
is provided below.</p>
<dl class="field-list simple">
<dt class="field-odd"><strong>a</strong></dt>
<dd class="field-odd"><p>In the pathway map, genes are colored according to their upregulation
(red) or downregulation (blue) in the contrast profile. Each pathway
is scored for the selected contrast profile and reported in the table
below.</p>
</dd>
<dt class="field-even"><strong>b</strong></dt>
<dd class="field-even"><p>Enrichment table. The table is interactive; enabling user to sort on
different variables (kegg id, pathway, logFC and meta q-values)
and select a pathway by clicking on the row in the table.</p>
</dd>
<dt class="field-odd"><strong>c</strong></dt>
<dd class="field-odd"><p>The KEGG activation matrix visualizes the activation levels of pathways
(or pathway keywords) across multiple contrast profiles. This facilitates
to quickly see and detect the similarities of certain pathways between
contrasts. The size of the circles correspond to their relative activation,
and are colored according to their upregulation (red) or downregulation
(blue) in the contrast profile. The matrix can be normalised from the
<em>Settings</em>.</p>
</dd>
</dl>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc6.1.png"><img alt="../_images/psc6.1.png" src="../_images/psc6.1.png" style="width: 100%;" /></a>
</figure>
</section>
<section id="go-graph">
<h3>GO graph<a class="headerlink" href="#go-graph" title="Permalink to this headline">¶</a></h3>
<p>In the <strong>GO</strong> panel, users can perform <a class="reference external" href="http://geneontology.org/">GO</a> analysis.
GO defines functional concepts/classes and their relationships as a hierarchical
graph.
The GO database provides a computational representation of the current knowledge
about roles of genes for many organisms in terms of molecular functions, cellular
components and biological processes. All the features described under the
<strong>KEGG pathway</strong> panel, such as scoring the gene sets and drawing an
activation-heatmap,
can be performed for the GO database under the GO graph tab. Instead of pathway
maps, an annotated graph structure provided by the GO database is potted for
every selected gene set.
Each output chart/table of the panel is describer below in detail.</p>
<dl class="field-list">
<dt class="field-odd"><strong>a</strong></dt>
<dd class="field-odd"><p>The structure of GO can be described in terms of a graph, where each
GO term is a node, and the relationships between the terms are edges
between the nodes. GO is loosely hierarchical, with ‘child’ terms being
more specialized than their ‘parent’ terms. The graph is interactive.
You can move the graph and zoom in using the mouse.
Under the graph <em>Settings</em>, users can select <code class="docutils literal notranslate"><span class="pre">Prune</span> <span class="pre">tree</span></code> to prune
the tree only with significant branches and <code class="docutils literal notranslate"><span class="pre">color</span> <span class="pre">custers</span></code> to
highlight clusters with different colors</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc6.2.a.png"><img alt="../_images/psc6.2.a.png" src="../_images/psc6.2.a.png" style="width: 35%;" /></a>
</figure>
</dd>
<dt class="field-even"><strong>b</strong></dt>
<dd class="field-even"><p>GO score table. The scoring of a GO term is performed by considering
the cumulative score of all terms from that term to the root node.
That means that GO terms that are supported by higher level terms
levels are preferentially scored.</p>
</dd>
<dt class="field-odd"><strong>c</strong></dt>
<dd class="field-odd"><p>The GO activation matrix visualizes the activation of GO terms
across conditions. From this figure, you can easily detect GO terms
that are consistently up/down across conditions. The size of the circles
correspond to their relative activation, and are colored according to
their upregulation (red) or downregulation (blue) in the contrast
profile. The matrix can be normalised from the <em>Settings</em>.</p>
</dd>
</dl>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc6.2.png"><img alt="../_images/psc6.2.png" src="../_images/psc6.2.png" style="width: 100%;" /></a>
</figure>
</section>
</section>
<section id="word-cloud">
<h2>Word Cloud<a class="headerlink" href="#word-cloud" title="Permalink to this headline">¶</a></h2>
<p>The <strong>WordCloud</strong> panel performs “keyword enrichment analysis”. It
computes enrichment of a selected keyword across all contrasts. Select
a keyword by clicking a word in the ‘Enrichment table’. Keyword
enrichment is computed by running GSEA on the enrichment score profile
for all contrasts. We defined the test set as the collection of
genesets that contain the keyword in the title/description.</p>
<section id="id3">
<h3>Input panel<a class="headerlink" href="#id3" title="Permalink to this headline">¶</a></h3>
<p>A tutorial about the module can be found under <code class="docutils literal notranslate"><span class="pre">Youtube</span></code> in the input panel.
Users can specify the contrast of their interest
in the <code class="docutils literal notranslate"><span class="pre">Contrast</span></code> settings. Under the main <em>Options</em>, users can
select <code class="docutils literal notranslate"><span class="pre">normalize</span> <span class="pre">activation</span> <span class="pre">matrix</span></code>  to normalize the activation matrix.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc6.4.0.png"><img alt="../_images/psc6.4.0.png" src="../_images/psc6.4.0.png" style="width: 30%;" /></a>
</figure>
</section>
<section id="main-panel">
<h3>Main panel<a class="headerlink" href="#main-panel" title="Permalink to this headline">¶</a></h3>
<p>The main panel consists of six different outputs:</p>
<dl class="field-list simple">
<dt class="field-odd"><strong>a</strong></dt>
<dd class="field-odd"><p>The <strong>Enrichment plots</strong> visualize the enrichment of the
selected keyword in the contrasts.  Black vertical bars
indicate the position of gene sets that contains the <em>keyword</em>
in the ranked list of enrichment scores. The curve in green
corresponds to the ‘running statistic’ of the keyword
enrichment score. The more the green ES curve is shifted to
the upper left of the graph, the more the keyword is enriched
in the first group. Conversely, a shift of the green ES curve
to the lower right, corresponds to keyword enrichment in the
second group.</p>
</dd>
<dt class="field-even"><strong>b</strong></dt>
<dd class="field-even"><p>The <strong>Word cloud plot</strong> visualizes the frequency/enrichment of
keywords for the data set. Select a keyword in the ‘Enrichment
table’. In the plot settings, users can exclude certain words
from the figure, or choose the color palette. The sizes of the
words are relative to the normalized enrichment score (NES)
from the GSEA computation. Keyword enrichment is computed by
running GSEA on the mean (squared) enrichment profile
(averaged over all contrasts). For each keyword, we defined
the ‘keyword set’ as the collection of genesets that contain
that keyword in the title/description.</p>
</dd>
</dl>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc6.4.1.png"><img alt="../_images/psc6.4.1.png" src="../_images/psc6.4.1.png" style="width: 30%;" /></a>
</figure>
<dl class="field-list simple">
<dt class="field-odd"><strong>c</strong></dt>
<dd class="field-odd"><p>The <strong>Word t-SNE</strong> plot visualizes the similarity of the
keywords that were found in the title/description of gene
sets. Keywords that are often found together in
title/descriptions are placed close together in the t-SNE. For
each keyword we computed enrichment using GSEA on the mean
(absolute) enrichment profiles (averaged over all
contrasts). Statistically significant gene sets (q&lt;0.05) are
colored in red. The sizes of the nodes are proportional to the
normalized enrichment score (NES) of the keyword. In the plot
settings, the user can choose between t-SNE and “Uniform
Manifold Approximation and Projection” (UMAP).</p>
</dd>
</dl>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc6.4.2.png"><img alt="../_images/psc6.4.2.png" src="../_images/psc6.4.2.png" style="width: 30%;" /></a>
</figure>
<dl class="field-list simple">
<dt class="field-odd"><strong>d</strong></dt>
<dd class="field-odd"><p>The <strong>Activation matrix</strong> visualizes the keywords enrichment
across the conditions. The size of the circles correspond to
their relative activation, and are colored according to their
upregulation (red) or downregulation (blue) in the contrast profile.</p>
</dd>
<dt class="field-even"><strong>e</strong></dt>
<dd class="field-even"><p>The <strong>Enrichment table</strong> summarizes the results from the
enrichment test for the tested keywords. The NES corresponds
to the normalized enrichment score from the GSEA analysis.</p>
</dd>
<dt class="field-odd"><strong>f</strong></dt>
<dd class="field-odd"><p>The <strong>Leading-edge table</strong> shows the geneset titles that have
contributed to the enrichment of the selected keyword.</p>
</dd>
</dl>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc6.4.png"><img alt="../_images/psc6.4.png" src="../_images/psc6.4.png" style="width: 100%;" /></a>
</figure>
</section>
</section>
<section id="drug-connectivity">
<h2>Drug connectivity<a class="headerlink" href="#drug-connectivity" title="Permalink to this headline">¶</a></h2>
<p>In the <strong>Drug Connectivity Map</strong> panel, users can correlate their signature with
more than 5000 known drug profiles from the
<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pubmed/29195078">L1000</a> database, as well as with drug
sensitivity profiles from the <a class="reference external" href="https://portals.broadinstitute.org/ctrp.v2.1/">CTRP v2</a>
and <a class="reference external" href="https://www.cancerrxgene.org/">GDSC</a> databases.
Additionally, a separate list of shRNA- and cDNA-perturebed datasets from the L1000 database
is also available (gene/L1000).</p>
<p>An activation-heatmap compares drug activation profiles across multiple contrasts.
This facilitates to quickly see and detect the similarities between contrasts
for certain drugs.</p>
<section id="id4">
<h3>Input panel<a class="headerlink" href="#id4" title="Permalink to this headline">¶</a></h3>
<p>A tutorial about the module can be found under <code class="docutils literal notranslate"><span class="pre">Youtube</span></code> in the input panel.
Users can specify the contrast of their interest
in the <code class="docutils literal notranslate"><span class="pre">Contrast</span></code> settings. Under <code class="docutils literal notranslate"><span class="pre">Analysis</span> <span class="pre">type</span></code> users can select from four
databases.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc6.3.0.png"><img alt="../_images/psc6.3.0.png" src="../_images/psc6.3.0.png" style="width: 30%;" /></a>
</figure>
</section>
<section id="id5">
<h3>Main panel<a class="headerlink" href="#id5" title="Permalink to this headline">¶</a></h3>
<p>The main panel consists of four outputs:</p>
<dl class="field-list">
<dt class="field-odd"><strong>a</strong></dt>
<dd class="field-odd"><p>The Drug Connectivity Map correlates your signature with profiles from the L1000
(activity/L1000 and gene/L1000), CTRP and GDSC databases.
It shows the top N=10 similar and opposite profiles by running
the GSEA algorithm on the contrast-drug profile correlation space.</p>
</dd>
<dt class="field-even"><strong>b</strong></dt>
<dd class="field-even"><p>Enrichment table. Enrichment is calculated by correlating
your signature with the profiles from the chosen
database. Because of multiple perturbation experiments for a
single small molecule, they are scored by running the GSEA algorithm on the
contrast-small molecule profile correlation space. In this way, we obtain a
single score for multiple profiles of a single small molecule. The table can be
customised via the table <em>Settings</em> to only show annotated drugs.</p>
</dd>
<dt class="field-odd"><strong>c</strong></dt>
<dd class="field-odd"><p>This plot visualizes the mechanism of action (MOA) across the enriched
drug profiles. On the vertical axis, the number of drugs with the same
MOA are plotted. You can switch to visualize between MOA or target gene.
Under the plots <em>Settings</em>, users can select the plot type of MOA
analysis: by class description (<code class="docutils literal notranslate"><span class="pre">drug</span> <span class="pre">class</span></code>) or by target gene
(<code class="docutils literal notranslate"><span class="pre">target</span> <span class="pre">gene</span></code>).</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc6.3.c.png"><img alt="../_images/psc6.3.c.png" src="../_images/psc6.3.c.png" style="width: 35%;" /></a>
</figure>
</dd>
<dt class="field-even"><strong>d</strong></dt>
<dd class="field-even"><p>The <strong>Activation matrix</strong> visualizes the correlation of small molecule
profiles with all available pairwise comparisons. The size of the
circles correspond to the strength of their correlation, and are
colored according to their positive (red) or
negative (blue) correlation to the contrast profile.
The matrix can be normalised from the <em>Settings</em>.</p>
</dd>
</dl>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc6.3.png"><img alt="../_images/psc6.3.png" src="../_images/psc6.3.png" style="width: 100%;" /></a>
</figure>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="mod3_DEGs.html" class="btn btn-neutral float-left" title="Expression" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="mod7_Signature.html" class="btn btn-neutral float-right" title="Signature" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright .</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>