<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Cell Profiling &mdash; Omics Playground  documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
    <link rel="shortcut icon" href="../_static/bigomics-logo-white-48px.png"/>
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Reanalyzing Public Datasets" href="../examples/examples.html" />
    <link rel="prev" title="Signature" href="mod7_Signature.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            Omics Playground
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Getting Started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../overview.html">Overview</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Basic Data Preparation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/dataprep.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/samples.html">Samples file</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/counts.html">Counts file</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/contrasts.html">Contrasts file (optional)</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/datachecks.html">Checks before uploading</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/uploadopg.html">Uploading your data in Omics Playground</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Advanced Data Preparation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/computepgx.html">What if I have many datasets?</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/dataprep_example.html">Preparing input files from raw data formats</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/geneset.html">Loading custom geneset files</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dataprep/dataprocess.html">Data processing inside Omics Playground</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Analysis Modules</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="outline.html">Outline</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod0_home.html">Home Panel</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod1_dataview.html">Data View</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod2_clustering.html">Clustering</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod3_DEGs.html">Expression</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod4_DEGSs.html">Enrichment</a></li>
<li class="toctree-l1"><a class="reference internal" href="mod7_Signature.html">Signature</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Cell Profiling</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#input-panel">Input panel</a></li>
<li class="toctree-l2"><a class="reference internal" href="#cell-type">Cell type</a></li>
<li class="toctree-l2"><a class="reference internal" href="#mapping">Mapping</a></li>
<li class="toctree-l2"><a class="reference internal" href="#markers">Markers</a></li>
<li class="toctree-l2"><a class="reference internal" href="#italk">iTALK</a></li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Examples</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../examples/examples.html">Reanalyzing Public Datasets</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Methods</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../methods.html">Methods</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">FAQ</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../faq.html">Frequently Asked Questions (FAQ)</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">About</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../info.html">Contact</a></li>
<li class="toctree-l1"><a class="reference internal" href="../info.html#citation">Citation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../info.html#license">License</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Omics Playground</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Cell Profiling</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/modules/mod9_CellProfiling.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="cell-profiling">
<span id="cellprofiling"></span><h1>Cell Profiling<a class="headerlink" href="#cell-profiling" title="Permalink to this headline">¶</a></h1>
<p>The <strong>Cell Profiling</strong> module is specifically developed for the
analysis and visualization of single-cell datasets. The main
applications are identification of immune cell types and
visualisations of markers, phenotypes, and proportions across the cells.</p>
<p>The <strong>Cell type</strong> panel infers the type of cells using computational deconvolution
methods and reference datasets from the literature.</p>
<p>The <strong>Mapping</strong> panel provides a visualization of the inferred cell types
matched to the phenotype variable of the data set, as well as a proportion plot
visualizing the interrelationships between two categorical variables
(so-called cross tabulation). This can be used to study the composition
of a sample by cell type, for example.</p>
<p>The <strong>Markers</strong> panel provides potential marker genes, which are the top genes
with the highest standard deviation within the expression data across the samples.
It also generates a plot mimicking the scatter plots used for gating in
flow cytometry analysis.</p>
<p>The <strong>iTALK</strong> consists of a computational approach from the literature
to characterize and illustrate intercellular communication signals in the
multicellular tumor ecosystem.</p>
<section id="input-panel">
<h2>Input panel<a class="headerlink" href="#input-panel" title="Permalink to this headline">¶</a></h2>
<p>Users can filter relevant samples in the <code class="docutils literal notranslate"><span class="pre">Filter</span> <span class="pre">samples</span></code> settings
under the the main <code class="docutils literal notranslate"><span class="pre">Options</span></code> in the input panel. They can also
specify to use a <code class="docutils literal notranslate"><span class="pre">default</span></code>  (tSNE) or <code class="docutils literal notranslate"><span class="pre">pca</span></code> layout for the figures.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc10.0.png"><img alt="../_images/psc10.0.png" src="../_images/psc10.0.png" style="width: 30%;" /></a>
</figure>
</section>
<section id="cell-type">
<h2>Cell type<a class="headerlink" href="#cell-type" title="Permalink to this headline">¶</a></h2>
<p>The <strong>Cell type</strong> profiling panel infers the type of cells using
computational deconvolution methods and reference datasets from the
literature.  In the plot settings menu, users can select the
reference dataset and the method for the cell type prediction in the
<code class="docutils literal notranslate"><span class="pre">reference</span></code> and <code class="docutils literal notranslate"><span class="pre">method</span></code> settings, respectively. Currently, we
have implemented a total of 7 methods (EPIC, DeconRNAseq, DCQ, I-NNLS,
NNLM, correlation-based and a meta-method) and 9 reference datasets to
predict immune cell types (4 datasets: LM22, ImmProt, DICE and
ImmunoStates), tissue types (2 datasets: HPA and GTEx), cell lines (2
datasets: HPA and CCLE) and cancer types (1 dataset: CCLE). Not all
methods or databases may be available for a dataset, the availability
depends on the pre-processing done. From the settings, users can also
sort plots by either probability or name and change the layout.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc10.1.0.png"><img alt="../_images/psc10.1.0.png" src="../_images/psc10.1.0.png" style="width: 30%;" /></a>
</figure>
<p>The cell type profiling plot is highlighted below.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc10.1.png"><img alt="../_images/psc10.1.png" src="../_images/psc10.1.png" style="width: 100%;" /></a>
</figure>
<p>It also produces phenotypes plots showing the distribution of the
phenotypes superimposed on the t-SNE clustering.
Often, we can expect the t-SNE distribution to be driven by the particular
phenotype that is controlled by the experimental condition or unwanted
batch effects. Users can select whether they want the group labels to be
plotted inside the figures or in a seperate legend in the``Legend``
under the <em>Settings</em>.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc10.1.1.png"><img alt="../_images/psc10.1.1.png" src="../_images/psc10.1.1.png" style="width: 100%;" /></a>
</figure>
</section>
<section id="mapping">
<h2>Mapping<a class="headerlink" href="#mapping" title="Permalink to this headline">¶</a></h2>
<p>The <strong>Mapping</strong> panel contains two plots.
To the right there is a plot representing the cell type mapping across all samples.
This plot can be customised via the <em>Settings</em> menu. Through it, users can change
the <code class="docutils literal notranslate"><span class="pre">plot</span> <span class="pre">type</span></code> between a dotmap and a heatmap, select the <code class="docutils literal notranslate"><span class="pre">reference</span></code> dataset,
select the analysis <code class="docutils literal notranslate"><span class="pre">method</span></code> and use <code class="docutils literal notranslate"><span class="pre">group</span> <span class="pre">by</span></code> to group samples by phenotypes.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc10.2.a.png"><img alt="../_images/psc10.2.a.png" src="../_images/psc10.2.a.png" style="width: 30%;" /></a>
</figure>
<p>The cell type mapping plot is shown below.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc10.2.png"><img alt="../_images/psc10.2.png" src="../_images/psc10.2.png" style="width: 100%;" /></a>
</figure>
<p>To the left, a proportion plot visualizes the overlap between two categorical variables.
This can be useful for bulk RNA datasets, as it provides useful information about
the proportion of different cell types in the samples. From the <em>Settings</em>, users
can select whwther to display the &lt;cell type&gt; (based on the chosen reference dataset)
or select one of the available phenotypes on the x- and y-axes of the plot. The number
of counts is also displayed on top.
By selecting a gene with <code class="docutils literal notranslate"><span class="pre">gene</span></code> they can also add an expression barplot for the gene.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc10.2.b.png"><img alt="../_images/psc10.2.b.png" src="../_images/psc10.2.b.png" style="width: 30%;" /></a>
</figure>
<p>The proportion plot (including a gene expression barplot) is shown below.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc10.2.1.png"><img alt="../_images/psc10.2.1.png" src="../_images/psc10.2.1.png" style="width: 100%;" /></a>
</figure>
</section>
<section id="markers">
<h2>Markers<a class="headerlink" href="#markers" title="Permalink to this headline">¶</a></h2>
<p>The <strong>Markers</strong> panel produces two outputs. The first output consists of 36 t-SNE
plots of the genes with the highest standard deviation that could represent
potential biomarkers. The red color shading is proportional to the (absolute)
expression of the gene in corresponding samples.</p>
<p>In the plot <em>Settings</em>, users can specify the <code class="docutils literal notranslate"><span class="pre">Level</span></code> of the marker analysis:
gene or gene set level. They can also restrict the analysis by selecting a particular
functional group in the <code class="docutils literal notranslate"><span class="pre">Feature</span> <span class="pre">set</span></code>, where genes are divided into 89 groups, such as
chemokines, transcription factors, genes involved in immune checkpoint inhibition, and so on.
In addition, it is possible to filter markers by a specific keywords in the <code class="docutils literal notranslate"><span class="pre">Filter</span></code> setting
and sort them by name or intensity.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc10.3.a.png"><img alt="../_images/psc10.3.a.png" src="../_images/psc10.3.a.png" style="width: 30%;" /></a>
</figure>
<p>An example of the t-SNE plots are highlighted below.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc10.3.png"><img alt="../_images/psc10.3.png" src="../_images/psc10.3.png" style="width: 100%;" /></a>
</figure>
<p>For each gene pairs combination, the panel also generates a cytometry-like plot (<strong>Cyto plot</strong>)
of samples. The aim of this feature is to observe the distribution of samples
in relation to the selected gene pairs. For instance, when applied to single-cell
sequencing data from immunological cells, it can mimic flow cytometry analysis and distinguish
T helper cells from other T cells by selecting the CD4 and CD8 gene combination.
Under the plot <em>Settings</em>, user can select their prefered genes on the x- and y-axes
in the <code class="docutils literal notranslate"><span class="pre">x-axis</span></code> and <code class="docutils literal notranslate"><span class="pre">y-axis</span></code>, respectively.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc10.3.b.png"><img alt="../_images/psc10.3.b.png" src="../_images/psc10.3.b.png" style="width: 30%;" /></a>
</figure>
<p>The Cyto plot is highlighted below.</p>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc10.3.1.png"><img alt="../_images/psc10.3.1.png" src="../_images/psc10.3.1.png" style="width: 100%;" /></a>
</figure>
</section>
<section id="italk">
<h2>iTALK<a class="headerlink" href="#italk" title="Permalink to this headline">¶</a></h2>
<p>The <strong>iTALK</strong> panel implements the
<a class="reference external" href="https://www.biorxiv.org/content/10.1101/507871v1">iTALK</a>
package from the literature. It is designed to profile and visualize the
ligand-receptor mediated intercellular cross-talk signals from single-cell
RNA sequencing data (scRNA-seq). iTALK uses a manually curated list of
ligand-receptor gene pairs further classified into 4 categories based on
the primary function of the ligand: cytokines/chemokines,
immune checkpoint genes, growth factors, and others.
Phenotype and ligand-receptor gene pairs can be selected from the
scrolldown menus on top of the Ligand-Receptor plot.</p>
<p>The panel produces three plots:</p>
<dl class="field-list simple">
<dt class="field-odd"><strong>a</strong></dt>
<dd class="field-odd"><p>The Ligand-Receptor plot visualizes the communication structure of
ligand-receptor genes as a circle plot. The width of the arrow represents
the expression level/log fold change of the ligand; while the width of
arrow head represents the expression level/log fold change of the
receptor. Different color and the type of the arrow stands for whether
the ligand and/or receptor are upregulated or downregulated.
Under the <em>Settings</em>, it is possible to select the number of top pairs
to be displayed in the <code class="docutils literal notranslate"><span class="pre">ntop</span> <span class="pre">pairs</span></code>.</p>
</dd>
<dt class="field-even"><strong>b</strong></dt>
<dd class="field-even"><p>The heatmap visualizes the expression level/log fold change of the
ligand/receptor genes. For further information, see
<a class="reference external" href="https://www.biorxiv.org/content/10.1101/507871v1">iTALK</a> R package.</p>
</dd>
<dt class="field-odd"><strong>c</strong></dt>
<dd class="field-odd"><p>The NetView plot visualizes the communication structure of ligand-receptor
genes as a graph. The colors represent different types of cells as a
structure and the width of edges represent the strength of the communication.
Labels on the edges show exactly how many interactions exist between two
types of cells. For further information, see
<a class="reference external" href="https://www.biorxiv.org/content/10.1101/507871v1">iTALK</a> R package.
Users can set the number of top genes to be plotted in the figure in the
<code class="docutils literal notranslate"><span class="pre">top</span> <span class="pre">genes</span></code> under the <em>Settings</em>.</p>
</dd>
</dl>
<figure class="align-center">
<a class="reference internal image-reference" href="../_images/psc10.4.png"><img alt="../_images/psc10.4.png" src="../_images/psc10.4.png" style="width: 100%;" /></a>
</figure>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="mod7_Signature.html" class="btn btn-neutral float-left" title="Signature" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="../examples/examples.html" class="btn btn-neutral float-right" title="Reanalyzing Public Datasets" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright .</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>